geopyspark/geotrellis/tile_layer.py

```diff
@@ -375,22 +406,63 @@ def reproject(geopysc,
                                          schema,
                                          json.dumps(tile_layer_metadata),
                                          dest_crs,
+                                         layout_extent,
+                                         tile_layout,
+                                         resample_method,
                                          match_layer_extent)
 
     (rdd, returned_schema) = (result._2()._1(), result._2()._2())
     ser = geopysc.create_tuple_serializer(returned_schema, value_type=TILE)
     returned_rdd = geopysc.create_python_rdd(rdd, ser)
 
-    return (returned_rdd, json.loads(result._3()))
+    return (result._1(), returned_rdd, json.loads(result._3()))
 
-def reproject_pyramid(geopysc,
-                      rdd_type,
-                      keyed_rdd,
-                      tile_layer_metadata,
-                      dest_crs,
-                      tile_size,
-                      resolution_threshold=0.1,
-                      match_layer_extent=False):
+def reproject_with_floating_layout(geopysc,
```

Comment:
It would be good to avoid all of these renames: `reproject_with_floating_layout`, `reproject_with_tile_layout`, `reproject_with_zoomed_layout` and have a single `reproject` method. I'm not sure I track why they are required.